ingestion/src/metadata/orm_profiler/processor/orm_profiler.py

```diff
@@ -170,6 +174,28 @@ def get_partition_details(
 
         return None
 
+    def get_profile_sample_query(self, table: Table) -> Optional[str]:
+        """Get sample query from the test definition. We first check
+        if the profiler workflow config file contains a `profile_sample_query`
+        in one of the test case. If not we'll check the table entity itself,
+        first checking if the table entity has a tableProfile object and then
+        if it has a `profileQuery` field.
+
+        Args:
+            table: table object
+        Returns
+            Optional[str]
+        """
+        if self.config.test_suite:
+            test_record = self.get_record_test_def(table)
+            if test_record:
+                return test_record.profile_sample_query
+
+        if table.tableProfile:
+            return table.tableProfile.profileQuery
```

Comment:
`tableProfile` is actually a time-series representation of the results of all profiling. IMO the `profileQuery` should be there as you added, but also as a root property of the table. That is from where we can pick up the data. On the other hand, the `tableProfile` will just inform how each profiling was executed.

Imagine the example with the `profileSample`. in `table.profileSample` we have the number we want to use, the latest one being set in the table. Then, in the tableProfile series we can see the evolution of that profileSample

Reply:
then note how in the `metadata_rest` from the orm_profiler we are updating the profileSample from the table Entity via `self.metadata.update_profile_sample`. Here we could do the same with the profileQuery:
1. we record it in the tableProfile to showcase how the metrics got computed
2. we update the table.profileQuery to have an up-to-date state of that property in the Entity

Reply:
Ah makes sense. I didn't have that timestamp nuance in regards to the `tableProfile`. Let me fix that.